### PR TITLE
[no_early_kickoff][Train] `ray.train.huggingface` restructure

### PR DESCRIPTION
**Pull request by Yard1**
## Why are these changes needed?

Renames `HuggingFaceTrainer`/`Predictor`/`Checkpoint` to `TransformersTrainer`/`Predictor`/`Checkpoint` for clarity and moves them to `ray.train.transformers`. A soft deprecation is introduced so that the old location and names still work.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ollie-iterators** [comment]
Documentation error:
/ray/doc/source/train/api/api.rst:222: WARNING: autosummary: failed to import train.huggingface.TransformersTrainer.restore

File: python/ray/train/huggingface/huggingface_predictor.py [hunk 1]
```diff
@@ -1,243 +1,21 @@
...
-try:
-    import torch
 
-    torch_get_gpus = torch.cuda.device_count
-except ImportError:
+@Deprecated(message=deprecation_msg)
+class HuggingFacePredictor(TransformersPredictor):
+    # Use __new__ as it is much less likely to be overriden
+    # than __init__
+    def __new__(cls: type):
```
Comment (woshiyyya) [inline python/ray/train/huggingface/huggingface_predictor.py:15]: same as above

**Review by woshiyyya — COMMENTED** [review]
LGTM. A few comments about the `__new__` methods of the deprecated classes.

File: python/ray/train/BUILD [hunk 2]
```diff
@@ -215,15 +215,6 @@ py_test(
 # Please keep these sorted alphabetically.
 # --------------------------------------------------------------------
 
-py_test(
```
[the remark this answers is not in the document]
Comment (Yard1) [reply python/ray/train/BUILD:218]: no, not sure what happened here. good catch!

**Review by woshiyyya — APPROVED** [review]

**Review by richardliaw — APPROVED** [review]
stamp

**Pull request merged**